en/includes/guides/authentication/conditional-auth/on-demand-silent-password-migration-template.md

```diff
@@ -0,0 +1,378 @@
+                'i18nkey': 'auth.fail.error'
+            });
+        }
+    });
+};
+
+
+/**
+ * This function will re-authenticate the user with the new password.
+ */
+var reAuthenticate = function() {
+
+    // Re-authenticate without prompting user input.
+    Log.info("Re-authenticating the user: " + userId + " with the new password.");
+
+    executeStep(1, {
+        authenticatorParams: {
+            common: {
+                'username': username,
+                'password': password
+            }
+        },
+    }, {
+        onSuccess: function(context) {
+            Log.info("Re-authentication successful for the user: " + userId + ".");
+        },
+        onFail: function() {
+            Log.info("Re-authentication failed for the user: " + userId + ".");
+
+            sendError(null, {
+                'status': 'Authentication failed',
+                'statusMsg': 'Please contact your administrator.',
+                'i18nkey': 'auth.fail.error'
+            });
+        }
+    });
+};
+```
+
+Replace the following parameters of the script with values relevant to your setup:
+
+<table>
+    <tr>
+        <th>start_authentication_endpoint</th>
+        <td>URL of the start authentication endpoint deployed in Choreo</td>
+    </tr>
+    <tr>
+        <th>polling_endpoint</th>
+        <td>URL of the polling endpoint deployed in Choreo</td>
+    </tr>
+    <tr>
+        <th>authentication_status_endpoint</th>
+        <td>URL of the authentication status endpoint deployed in Choreo</td>
+    </tr>
+    <tr>
+        <th>consumer_key</th>
+        <td>The consumer key of your Choreo application</td>
+    </tr>
+    <tr>
+        <th>consumer_secret</th>
+        <td>The consumer secret of your Choreo application</td>
+    </tr>
+    <tr>
+        <th>asgardeo_token_endpoint</th>
+        <td>Token endpoint of your Asgardeo organization. For example: <code>https://api.asgardeo.io/t/{organization_name}/oauth2/token</code>
+</td>
+    </tr>
+</table>
+
+## How it works
+
+Let's look at how the above conditional authentication scripts works.
+
+1. The first authentication step (Username & Password) is initiated with the `executeStep(1, ..)` function. Based on its status, one of the following happens.
+
+    - If the entered credentials match, the user's password is already migrated. Hence, the `onSuccess` function will be called and the user will be authenticated.
+    
+    - If the user's credentials don't match, either the credentials are incorrect or the password may not have be migrated yet. Hence, the `onFail` callback function is called and the script continues.
+
+2. If the `onFail` function is called, the script will next try to locate a unique user in the system using the `resolveAndInitUser` function. 
+
+    !!! note "Adjust script for alternate login identifiers"
+
+        If your organization does not use alternate login identifiers, uncomment the following lines in the `resolveAndInitUser` function.
```

Comment:
```suggestion
        If your organization does not use alternate login identifiers, comment the following lines in the `resolveAndInitUser` function.
```

We should call `resolveMultiAttributeLoginIdentifier` function only when alternative login is enabled